sections/multicolumn.liquid

```diff
@@ -81,17 +81,29 @@
                       style="padding-bottom: {{ 1 | divided_by: highest_ratio | times: 100 }}%;"
                     {% endif %}
                   >
-                    {%- capture sizes -%}
-                      (min-width: 990px) {% if section.blocks.size <= 2 %}710px{% else %}550px{% endif %}, (min-width:
-                      750px) {% if section.blocks.size == 1 %}710px{% else %}550px{% endif %}, calc(100vw - 30px)
-                    {%- endcapture -%}
+                    {%- liquid
+                      assign number_of_columns = section.settings.columns_desktop
+                      assign number_of_columns_mobile = section.settings.columns_mobile
+                      assign max_image_width = settings.page_width | times: 2
+                      if section.settings.image_width == 'half'
+                        assign image_width = 0.5
+                      elsif section.settings.image_width == 'third'
+                        assign image_width = 0.33
+                      else
+                        assign image_width = 1
+                      endif
+                    -%}
+                    {% capture sizes %}
+                      (min-width: 990px) {{ settings.page_width | times: image_width | divided_by: number_of_columns | round: 0 }}px,
+                      calc(100vw * {{ image_width }} / {{ number_of_columns_mobile }})
+                    {% endcapture %}
                     {{
                       block.settings.image
-                      | image_url: width: 1420
+                      | image_url: width: max_image_width
```

Comment:
I'm okay to hardcode 3200 as the last one instead of 3250 or replace the last one after 3000 with `max_image_width`. Thoughts?

Reply:
I don't feel strongly. I could see breaking the arbitrary increment amount for the last value when we know the actual maximum we'll need. So we could update both to be 3200, rather than using 3250 just because we're incrementing the others by 250.